### ISSUE TITLE
Shibboleth.pm - 025010a should be reverted

### ISSUE THREAD

**Issue by jmacdone** [openwebwork/webwork2#1844]
This change from https://github.com/openwebwork/webwork2/commit/025010a5b077b201eda13451877979ebe8875d31
```
- my $user_id = $ENV{$ce->{shibboleth}{mapping}{user_id}};
+ my $user_id = $ce->{shibboleth}{mapping}{user_id};
```
causes `$user_id` to be assigned name of the environment variable, rather than the desired value.

For example, from v2.16:
```
[Tue Nov 29 09:25:38.624653 2022] WeBWorK::Authen::Shibboleth::get_credentials: user_id is user_id
```
and after (manually) reverting, we get
```
[Tue Nov 29 10:32:38.20267 2022] WeBWorK::Authen::Shibboleth::get_credentials: user_id is jmacdone
```

**Comment by jmacdone** [comment]
FWIW, in the above example, the sysadmin is extracting what is typically referred to as `uid` instead as `user_id` and had `localOverrides.conf` set accordingly.

    jmacdone@webwork2:/etc/shibboleth$ grep user_id attribute-map.xml
    <Attribute name="urn:oid:0.9.2342.19200300.100.1.1"  id="user_id" />

and

    jmacdone@webwork2:/opt/webwork/webwork2/conf$ grep "'user_id'" localOverrides.conf
    $shibboleth{mapping}{user_id} = 'user_id';

**Comment by taniwallach** [comment]
@jmacdone - Thanks for reporting this. It certainly looks like a serious bug, and something to fix for WW 2.17 using a hot fix. I just created https://github.com/openwebwork/webwork2/pull/1845

@xcompass @mgage @drdrew42 @drgrice1 Could one of you take a look?

The problematic changes got in to the system via https://github.com/openwebwork/webwork2/pull/1056.

The Apache `mod_shib` documentation https://wiki.sunet.se/display/SWAMID/1.2+Configuring+Apache+Web+Server+to+use+Shibboleth  indicates that the data provided by the Identity Provider is intended to be accessed via "HTTP environment variables", namely via `$ENV` as was done, and you want to fix in one place.

Given that, it seems to me that all the changes from https://github.com/openwebwork/webwork2/pull/1056 are not correct. As I see it the changes to the conditionals are not correct either, as the test was almost certainly intended to check that `mod_shib` had handled the SSO authentication step and added the relevant data the `$ENV` from which webwork could get the data. There seems little value in just testing that the course environment was provided configuration settings with the names of the environment variables expected.

I don't think we can cleanly revert the old commit, as later changes were made to the file. Instead a new commit as a hotfix to main should just undo the bad changes, which is what I just did in https://github.com/openwebwork/webwork2/pull/1845 .

Regarding develop, more work will be needed as Apache+mod_perl has been replaced by Mojolicious. The method of authenticating via Shibboleth will apparently need to be redone to take into account either Apache or nginx being used as a reverse proxy. @jmacdone - your help in trying to get Shibboleth to work with develop (which will eventually become WW 2.18) would be greatly appreciated.
